avroparquet/src/main/scala/akka/stream/alpakka/avroparquet/impl/AvroParquetFlow.scala

```diff
@@ -0,0 +1,45 @@
+package akka.stream.alpakka.avroparquet.impl
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import org.apache.avro.generic.GenericRecord
+import org.apache.parquet.hadoop.ParquetWriter
+
+private[avroparquet] class AvroParquetFlow(writer: ParquetWriter[GenericRecord])
+    extends GraphStage[FlowShape[GenericRecord, GenericRecord]] {
+
+  val in: Inlet[GenericRecord] = Inlet("AvroParquetSink.in")
+  val out: Outlet[GenericRecord] = Outlet("AvroParquetSink.out")
+  override val shape: FlowShape[GenericRecord, GenericRecord] = FlowShape.of(in, out)
+
+  override def createLogic(attr: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) {
+      setHandler(
+        in,
+        new InHandler {
+
+          override def onUpstreamFinish(): Unit = {
+            //super.onUpstreamFinish()
+            writer.close()
+            completeStage()
+          }
+
+          override def onUpstreamFailure(ex: Throwable): Unit = {
+            super.onUpstreamFailure(ex)
+            writer.close()
```

Comment:
Please free resources in `postStop`. See #277

Reply:
Done.